src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -90,12 +91,76 @@ private static IEnumerable<TextEdit> FormatRazor(FormattingContext context, Razo
     {
-        return TryFormatFunctionsBlock(context, edits, source, node) ||
+        // We only want to run one of these
+        _ = TryFormatFunctionsBlock(context, edits, source, node) ||
             TryFormatCSharpExplicitTransition(context, edits, source, node) ||
             TryFormatHtmlInCSharp(context, edits, source, node) ||
-            TryFormatComplexCSharpBlock(context, edits, source, node);
+            TryFormatComplexCSharpBlock(context, edits, source, node) ||
+            TryFormatSectionBlock(context, edits, source, node);
+    }
+
+    private static bool TryFormatSectionBlock(FormattingContext context, List<TextEdit> edits, RazorSourceDocument source, SyntaxNode node)
+    {
+        // @section Goo {
+        // }
+        //
+        // or
+        //
+        // @section Goo
+        // {
+        // }
+        if (node is CSharpCodeBlockSyntax directiveCode &&
+            directiveCode.Children is [RazorDirectiveSyntax directive] &&
+            directive.DirectiveDescriptor?.Directive == SectionDirective.Directive.Directive &&
+            directive.Body is RazorDirectiveBodySyntax { CSharpCode: { } code })
+        {
+            var children = code.Children;
+            if (TryGetWhitespace(children, out var whitespaceBeforeSectionName, out var whitespaceAfterSectionName))
+            {
+                // For whitespace we normalize it differently depending on if its multi-line or not
+                FormatWhitespaceBetweenDirectiveAndBrace(whitespaceBeforeSectionName, directive, edits, source, context);
+                FormatWhitespaceBetweenDirectiveAndBrace(whitespaceAfterSectionName, directive, edits, source, context);
+
+                return true;
+            }
+            else if (children.TryGetOpenBraceToken(out var brace))
+            {
+                // If there is no whitespace at all we normalize to a single space
+                var start = brace.GetRange(source).Start;
+                var edit = new TextEdit
+                {
+                    Range = new Range { Start = start, End = start },
+                    NewText = " "
+                };
+                edits.Add(edit);
+
+                return true;
+            }
+        }
+
+        return false;
+
+        static bool TryGetWhitespace(SyntaxList<RazorSyntaxNode> children, [NotNullWhen(true)] out CSharpStatementLiteralSyntax? whitespaceBeforeSectionName, [NotNullWhen(true)] out UnclassifiedTextLiteralSyntax? whitespaceAfterSectionName)
+        {
+            // If there is whitespace between the directive and the section name, and the section name and the brace, they will be in the first child
+            // and third child of the 6 total children
+            whitespaceBeforeSectionName = null;
+            whitespaceAfterSectionName = null;
+            if (children.Count == 6 &&
```

Comment:
This seems hacky but that's probably just the Razor formatter in general 😩

Reply:
Yeah, pretty much. It's more of a comment on the syntax tree really. In a future world this would be a `SectionBlockSyntax` or something, so we'd know 100% what we're doing, instead of using these odd heuristics